pachinko/time_period_step_agent.py

```diff
@@ -61,37 +61,45 @@ def act(self, previous_observation: np.ndarray) -> Tuple[Any, dict]:
         return action, {}
 
 
-class Periodic:
-    """Agent that maintains and optimises on separate conversion rates for each step in period."""
+# A list containing a tuple of (num_successes, timestep) for each time an action was picked,
+# representing the full history of that action's conversions
+ConversionHistory = List[Tuple[int, int]]
+
+
+class BasePeriodic:
+    """Base class for agents that optimises on separate conversion rates for each step in period.
+
+    Args:
+        env: the env the agent will be run on (so that the agent knows the actions to choose from)
+        period_length: the length defining a repeating period, where each step will have a separate
+            conversion rate. eg for a weekly periodicity, where each weekday has its own conversion
+            rate, use period_length=7
+    """
 
     def __init__(self, env: Any, period_length: int):
         self.action_space = env.action_space
-        # List of dicts of actions mapped to tuple of (num_trails, num_successes) for
-        # action period idx
-        self.conversion_counts: List[Dict[Any, List[int]]] = [
-            {action: [0, 0] for action in range(self.action_space.n)} for _ in range(period_length)
+        # List of dicts of actions mapped to their conversion history, for each action period idx
+        self.conversions: List[Dict[Any, ConversionHistory]] = [
+            {action: [] for action in range(self.action_space.n)} for _ in range(period_length)
         ]
-        # Set conversion rate to infinity for unchosen actions
-        # to ensure all actions are tested at least once
+        # Set conversion rate to infinity for unchosen actions to ensure all actions are tested at
```

Comment:
nit: small bit of imprecision in comment that could cause a confusion: we're not "set"ting the conversion rate (that's set by the env), we are setting the agents initial belief in the conversion rate. Maybe just adding "beliefs"/"estimates"/"inferences" after the word "rate" would help.